network/l2.c

```diff
@@ -5,22 +5,70 @@
 #include "bm_os.h"
 #include "util.h"
 
-#define ethernet_packet_size_byte 14
+/* We store the egress port for the RX device and the ingress port of the TX device
+   in bytes 5 (index 4) and 6 (index 5) of the SRC IPv6 address. */
+#define egress_port_idx 4
+#define ingress_port_idx 5
+
+// Ethernet Header Sizes
+#define ethernet_destination_size_bytes 6
+#define ethernet_src_size_bytes 6
+#define ethernet_type_size_bytes 2
+
+// IPV6 Header Sizes
 #define ipv6_version_traffic_class_flow_label_size_bytes 4
 #define ipv6_payload_length_size_bytes 2
 #define ipv6_next_header_size_bytes 1
 #define ipv6_hop_limit_size_bytes 1
 #define ipv6_source_address_size_bytes 16
 #define ipv6_destination_address_size_bytes 16
 
+// UDP Header Sizes
+#define udp_src_size_bytes 2
+#define udp_destination_size_bytes 2
+#define udp_length_size_bytes 2
+#define udp_checksum_size_bytes 2
+
+// Ethernet Offsets
+#define ethernet_destination_offset (0)
+#define ethernet_src_offset                                                    \
+  (ethernet_destination_offset + ethernet_destination_size_bytes)
+#define ethernet_type_offset (ethernet_src_offset + ethernet_src_size_bytes)
+
+// Ethernet Getters
+#define ethernet_get_type(buf) (uint8_to_uint16(&buf[ethernet_type_offset]))
+
+// IPV6 Offsets
+#define ipv6_version_traffic_class_flow_label_offset                           \
+  (ethernet_type_offset + ethernet_type_size_bytes)
+#define ipv6_payload_length_offset                                             \
+  (ipv6_version_traffic_class_flow_label_offset +                              \
+   ipv6_version_traffic_class_flow_label_size_bytes)
+#define ipv6_next_header_offset                                                \
+  (ipv6_payload_length_offset + ipv6_payload_length_size_bytes)
+#define ipv6_hop_limit_offset                                                  \
+  (ipv6_next_header_offset + ipv6_next_header_size_bytes)
```

Comment:
Don't know if there is a better spot to place all of these as they are generic. As of right now only l2 uses them.

Reply:
Interesting, yeah — I think since they're only used in l2, then here is the right place. We can move them elsewhere in the future if they're ever needed more publicly.